### PR TITLE
copr: Vectorize exp.

### PR DESCRIPTION
**Pull request by cireu**
Signed-off-by: Zhu Zihao <all_but_last@163.com>

PCP #5751 

###  What have you changed?

 Vectorize `exp` 

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:

-  unit test
###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

**Comment by cireu** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/32

**Comment by cireu** [comment]
@sre-bot  /run-all-tests

**Comment by cireu** [comment]
@sre-bot  /run-all-test

**Comment by cireu** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/32

File: components/tidb_query/src/rpn_expr/impl_math.rs [hunk 1]
```diff
@@ -253,6 +253,22 @@ fn sqrt(arg: &Option<Real>) -> Result<Option<Real>> {
     }))
 }
 
+#[inline]
+#[rpn_fn]
+pub fn exp(arg: &Option<Real>) -> Result<Option<Real>> {
+    match arg {
+        Some(x) => {
+            let ret = x.exp();
+            if ret.is_infinite() || ret.is_nan() {
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_math.rs]: Please handle overflow with an EvalContext

Comment (cireu) [reply components/tidb_query/src/rpn_expr/impl_math.rs]: It doesn't work, and break push down test, here is a similar example https://github.com/tikv/tikv/pull/5873#issuecomment-553829633

**Pull request merged**